module/core/derive_tools/src/reflect/axiomatic.rs

```diff
@@ -277,6 +289,46 @@ pub( crate ) mod private
     }
   }
 
+  ///
+  /// Collection descriptor
+  ///
+  #[ derive( PartialEq, Default, Clone ) ]
+  pub struct CollectionDescriptor< I : Instance >
+  {
+    /// Container length.
+    pub len : usize,
+    /// Container keys.
+    pub keys : Option< Vec< primitive::Primitive > >,
```

Comment:
why? information about type of elements is known at compile time. field keys is redundant!